### ISSUE TITLE
[a11y] Images with no text alternative should still have alt=""

### ISSUE THREAD

**Issue by rshipp** [ckeditor/ckeditor5#8115]
## 📝 Provide a description of the improvement

_How the feature works now and what you'd like to change_?

Currently, if an image does not have a text alternative, the `img` element is generated in the data view with no `alt` attribute. According to the HTML spec for "[purely decorative images](https://html.spec.whatwg.org/multipage/images.html#a-purely-decorative-image-that-doesn't-add-any-information)", and accessibility standards, all `img` elements must have an `alt` attribute, but that attribute should be the empty string `alt=""` when the image does not add any information that would be described in the text alternative.

While the current code maintains an empty alt tag if it is added manually and loaded in through the data view, it does not add the empty alt tag by default to newly inserted images. I think this could be fixed by explicitly setting the `alt` attribute in [insertImage](https://github.com/ckeditor/ckeditor5/blob/91075a303e5b003b3a82fd96f57a651b6a3c95a9/packages/ckeditor5-image/src/image/utils.js#L82) if it's not passed in - but I'm not sure if that's the best place to do it. I can pass in an empty attribute when I call the insert command, and that will fix this in our editor, but it seems better if CKEditor were to generate accessible HTML by default, without me having to know this.

## 📃 Other details

* `ckeditor5-image` plugin
* tested on v19.0.0

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
Thanks for reporting this. We worked on it (#7405) but apparently, the patch was only partial.

**Comment by Reinmar** [comment]
To reproduce this:

```javascript
editor.setData('<img src=x>');

editor.getData(); // -> "<figure class="image"><img src="x"></figure>"
```

The `alt` attribute should be present in the above scenario.

**Comment by mgifford** [comment]
Having an explicit button to mark as decorative is a good practice. [A11yfirst](https://a11yfirst.gitlab.io/custom/a11yfirst.html) did this with CKEditor 4.

**Comment by CKEditorBot** [comment]
There has been no activity on this issue for the past year. We've marked it as stale and will close it in 30 days. We understand it may still be relevant, so if you're interested in the solution, leave a comment or reaction under this issue.
